nuxeo-routing-core/src/test/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriterTest.java

```diff
@@ -0,0 +1,505 @@
+        assertTaskDefaultProperties(json);
+
+        JsonNode delegatedActorsNode = json.get("delegatedActors").getNode();
+        assertEquals(1, delegatedActorsNode.size());
+
+        JsonNode actor = delegatedActorsNode.get(0);
+        assertThat(actor.size(), greaterThan(1));
+        assertTrue(actor.has("entity-type"));
+        assertEquals("user", actor.get("entity-type").textValue());
+        assertTrue(actor.has("id"));
+        assertEquals(session.getPrincipal().getName(), actor.get("id").textValue());
+    }
+
+    @Test
+    public void shouldFetchDelegatedUserPropertiesWhenActorsFetcherIsProvidedAndPrefixedDelegatedActorsExist()
+            throws IOException {
+        String workflowModel = "SerialDocumentReview";
+        String workflowInstanceId = documentRoutingService.createNewInstance(workflowModel, Collections.emptyList(),
+                Collections.emptyMap(), session, true);
+
+        List<Task> tasks = documentRoutingService.getTasks(null, null, workflowInstanceId, workflowModel, session);
+        assertEquals(1, tasks.size());
+
+        tasks.get(0).setDelegatedActors(Arrays.asList("user:Administrator", "group:administrators"));
+
+        JsonAssert json = jsonAssert(tasks.get(0), RenderingContext.CtxBuilder.fetch("task", "actors").get());
+        assertTaskDefaultProperties(json);
+
+        JsonNode delegatedActorsNode = json.get("delegatedActors").getNode();
+        assertEquals(2, delegatedActorsNode.size());
+
+        JsonNode actor = delegatedActorsNode.get(0);
+        assertThat(actor.size(), greaterThan(1));
+        assertTrue(actor.has("entity-type"));
+        assertEquals("user", actor.get("entity-type").textValue());
+        assertTrue(actor.has("id"));
+        assertEquals(session.getPrincipal().getName(), actor.get("id").textValue());
+
+        JsonNode group = delegatedActorsNode.get(1);
+        assertThat(group.size(), greaterThan(1));
+        assertTrue(group.has("entity-type"));
+        assertEquals("group", group.get("entity-type").textValue());
+        assertTrue(group.has("id"));
+        assertEquals("administrators", group.get("id").textValue());
+    }
+
+    @Test
+    public void shouldOnyWriteUserIdWhenInitiatorFetcherIsNotProvided() throws IOException {
+        String workflowModel = "SerialDocumentReview";
+        String workflowInstanceId = documentRoutingService.createNewInstance(workflowModel, Collections.emptyList(),
+                Collections.emptyMap(), session, true);
+
+        List<Task> tasks = documentRoutingService.getTasks(null, null, workflowInstanceId, workflowModel, session);
+        assertEquals(1, tasks.size());
+
+        JsonAssert json = jsonAssert(tasks.get(0));
+        assertTaskDefaultProperties(json);
+
+        JsonNode initiatorNode = json.get("workflowInitiator").getNode();
+        assertEquals(JsonNodeType.STRING, initiatorNode.getNodeType());
```

Comment:
`initiatorNode.isTextual()`?

Reply:
Yes, it could be assertTrue(initiatorNode.isTextual())